src/provider.ts

```diff
@@ -1,23 +1,58 @@
 export default (host: string) => {
+    const apiCall = async (request: any) => {
+        return await window.top.fetch(`${host}/v1/api/blockchain`, {
+            method: "POST",
+            cache: "no-cache",
+            credentials: "include",
+            keepalive: true,
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify(request),
+        });
+    };
+
     return {
         request: async function (request: any) {
             try {
+                console.log("request", request);
+                // First, we call wallet_getPermissions to check if user has given permissions or not
+                let response = await apiCall({
+                    method: "wallet_getPermissions",
+                });
+                let data = await response.json();
+                console.log("wallet_getPermissions", data);
+                if (
+                    !data.data ||
```

Comment:
Hey Jatin, just a heads-up, there's a PR I just opened on Point Node that will change the format of the response of all rpc requests, including `wallet_gerPermission`. The change is to make our API compatible with what other libraries, such as web3.js, expect.

In practice, this means that the response is simpler (no more `data.data`). This is an example of a response to `wallet_getPermissions`:

```json
{
    "id": "bc7362eec18e2ae4ab77f54fab9f0215",
    "invoker": "https://mydapp.point",
    "account": "0x916f8E7566Dd63D7c444468CaDeA37e80f7F8048",
    "parentCapabilities": [
        "eth_sendTransaction"
    ]
}
```

Reply:
Thanks for the heads up @germanviescas, I'll make the required changes.